packages/cli/src/UserManagement/PermissionChecker.ts

```diff
@@ -39,24 +39,21 @@ export class PermissionChecker {
 
 		if (user.hasGlobalScope('workflow:execute')) return;
 
+		const isSharingEnabled = this.license.isSharingEnabled();
+
 		// allow if all creds used in this workflow are a subset of
 		// all creds accessible to users who have access to this workflow
 
 		let workflowUserIds = [userId];
 
-		if (workflow.id && this.license.isSharingEnabled()) {
-			const workflowSharings = await this.sharedWorkflowRepository.find({
-				relations: ['workflow'],
-				where: { workflowId: workflow.id },
-				select: ['userId'],
-			});
-			workflowUserIds = workflowSharings.map((s) => s.userId);
+		if (workflow.id && isSharingEnabled) {
+			workflowUserIds = await this.sharedWorkflowRepository.getSharedUserIds(workflow.id);
 		}
 
-		const credentialSharings =
-			await this.sharedCredentialsRepository.findOwnedSharings(workflowUserIds);
-
-		const accessibleCredIds = credentialSharings.map((s) => s.credentialsId);
+		const accessibleCredIds = await this.sharedCredentialsRepository.getCredentialIdsForUserAndRole(
+			workflowUserIds,
+			isSharingEnabled ? ['credential:owner', 'credential:user'] : ['credential:owner'],
+		);
```

Comment:
Is this a leaky abstraction?

If `sharedCredentialsRepository.getAccessibleCredentialIds` finds by cred owner and cred user, then I'd've expected for
- there to be a `sharedCredentialsRepository.getOwnedCredentialIds` and 
- there to be a `OwnershipService.getExecutableCredentialIds` that calls `sharedCredentialsRepository.getAccessibleCredentialIds` or `sharedCredentialsRepository.getOwnedCredentialIds` depending on license state.